configbp/config.go

```diff
@@ -0,0 +1,69 @@
+// Package configbp parses configurations per the baseplate spec.
+package configbp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+// ExpandConfig performs baseplate-conformant configuration expansion.
+//
+// Currently, this includes expanding $FOO and ${FOO} with their respective environment variables.
+func ExpandConfig(rawConfig string) string {
+	expanded := os.ExpandEnv(rawConfig)
+	if expanded != rawConfig {
+		log.Debugf("Expanded configuration:\n%s", expanded)
+	}
+	return expanded
+}
+
+// ParseFile parses configuration from the file at the given path.
+//
+// Environment variables (e.g. $FOO and ${FOO}) are substituted from the environment before parsing.
+// The configuration is parsed into the target, which will typically be a pointer to a struct.
+//
+// Decoding is performed based on file extension:
+//   - YAML (i.e. ".yaml") files are decoded according to ParseYAML.
+func ParseFile(path string, ptr interface{}) error {
+	f, err := os.Open(path)
+	if err != nil {
+		return err // contains filename
+	}
+	defer f.Close() // safe to blindly close read-only files
+
+	switch ext := filepath.Ext(path); ext {
+	case ".yaml":
```

Comment:
what if the file is ending in `.yml`, which I believe is also widely used for yaml files?

since we only support one format that is yaml, I don't really think this check is really needed.

Reply:
This helps avoid accidents like the one you described in the baseplate.spec conversation :)

also, added .yml and downcasing.